test/types/awssigv4signer.test-d.ts

```diff
@@ -21,7 +21,11 @@ const mockCreds = {
 const mockRegion = 'us-west-2';
 
 {
-  const AwsSigv4SignerOptions = { credentials: mockCreds, region: mockRegion };
+  const AwsSigv4SignerOptions = {
+    getCredentials: Promise.resolve(mockCreds),
```

Comment:
`getCredentials` will not always be part of the input, how do we test types here?

Reply:
good highlight I think it should be something like that (but probably need to be translated to ts)
```suggestion
    getCredentials: (cb) => cb(null, mockCreds),
```
I'm not a typescript user so I don't have any clue and I will require some help with that, if that's really needed.
Disregarding the types, the code at runtime does test it + there is units test.

Reply:
updated but I would need some help with this to make typescript useful in this case.
the thing here is that all parameters are depending on which version of aws sdk you are using.
for aws v3 region and credentials are required.
for aws v2 only getCredentials is required (actually an AWS.Config type is required), region is optional as it is resolved to the default environment or profile region